docs/docs/adrs/adr-006-denom-dos-fixes

```diff
@@ -0,0 +1,53 @@
+---
+sidebar_position: 2
+title: ADR Template
+---
+# ADR 006: Denom DOS fixes
```

Comment:
Rename to 004 as it's already in main, while both https://github.com/cosmos/interchain-security/pull/909 and https://github.com/cosmos/interchain-security/pull/897 are work in progress.